src/physics_swe.c

```diff
@@ -267,15 +267,16 @@ static PetscErrorCode RHSFunctionForInternalEdges(RDy rdy, Vec F, AMaxDiagnostic
       PetscReal hl = x_ptr[l * ndof + 0];
       PetscReal hr = x_ptr[r * ndof + 0];
 
-      if (amax_vec_int[ii] > amax_diags->max_wave_speed) {
-        amax_diags->max_wave_speed = amax_vec_int[ii];
-        amax_diags->global_edge_id = edges->global_ids[ii];
-      }
-
       if (!(hr < tiny_h && hl < tiny_h)) {
         PetscReal areal = cells->areas[l];
         PetscReal arear = cells->areas[r];
 
+        PetscReal cnum = amax_vec_int[ii] * edge_len / fmin(areal, arear) * rdy->dt;
+        if (cnum > courant_num_diags->max_courant_num) {
+          courant_num_diags->max_courant_num = cnum;
+          courant_num_diags->global_edge_id  = edges->global_ids[ii];
```

Comment:
How about save the cell id as well? Given the edge id, the max Courant number may occur in either the left cell or the right cell.